testing/iframe.js

```diff
@@ -203,10 +203,8 @@ export function createIframePromise(opt_runtimeOff, opt_beforeLayoutCallback) {
         doc: iframe.contentWindow.document,
         iframe: iframe,
         addElement: function(element) {
-          iframe.contentWindow.document.getElementById('parent')
-              .appendChild(element);
-          // Wait for mutation observer to fire.
-          return new Timer(window).promise(16).then(() => {
+          const iWin = iframe.contentWindow;
+          const p = onInsert(iWin).then(() => {
```

Comment:
We could use `waitForChild`, but I think it is better to keep this self contained. The custom check function would be almost as complicated as the function here.